api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -185,4 +188,31 @@ public CohortDefinition getPatientsWithEnrollment(){
             "locationList", "locations"
         );
     }
+
+    @DocumentedDefinition("patientsWithState")
+    public CohortDefinition getPatientsWithState() {
+        PatientStateCohortDefinition cd = new PatientStateCohortDefinition();
+        cd.addParameter(new Parameter("states", "reporting.parameter.states", ProgramWorkflowState.class, List.class, null));
+        cd.addParameter(new Parameter("startedOnOrAfter", "reporting.parameter.startDate", Date.class));
+        cd.addParameter(new Parameter("startedOnOrBefore", "reporting.parameter.endDate", Date.class));
+        cd.addParameter(new Parameter("endedOnOrAfter", "reporting.parameter.startDate", Date.class));
+        cd.addParameter(new Parameter("endedOnOrBefore", "reporting.parameter.endDate", Date.class));
+        return new MappedParametersCohortDefinition(cd,
+            "states", "states",
+            "startedOnOrAfter", "enrolledOnOrAfter",
+            "startedOnOrBefore", "enrolledOnOrBefore",
+            "endedOnOrAfter", "completedOnOrAfter",
+            "endedOnOrBefore", "completedOnOrBefore"
+        );
```

Comment:
I don't see the value in changing these parameter names, and do not think these new names are more accurate.  I would simply return the PatientStateCohortDefinition, and remove the use of the MappedParametersCohortDefinition entirely.